src/formatter/helpers/gherkin_document_parser_spec.ts

```diff
@@ -0,0 +1,1294 @@
+import { describe, it, beforeEach } from 'mocha'
+import { expect } from 'chai'
+import {
+  getGherkinExampleRuleMap,
+  getGherkinScenarioLocationMap,
+  getGherkinScenarioMap,
+  getGherkinStepMap,
+} from './gherkin_document_parser'
+
+describe('GherkinDocumentParser', () => {
+  describe('with a Background and Scenario', () => {
+    let sourceData
+    beforeEach(() => {
+      // Arrange
+      sourceData = {
```

Comment:
Really don't like this as we are hardcoding some of gherkin's internal structure in a test that would need to be updated if gherkin changes something. 

Instead of hardcoding it, can we generate this by having the gherkin library parse some content?

Also when asserting the output, can we have maybe only one test assert the full output once to show structure and then later tests can assume the structure is preserved and perform simpler assertions?

Lastly, for general test structure I'm leaning towards no longer using beforeEach and instead just making helper functions so everything you need to know about the test is in the block (instead of having to find beforeEach blocks that setup some state). Thoughts on that?

Reply:
> Instead of hardcoding it, can we generate this by having the gherkin library parse some content?

Yeah that's much better, will do

> can we have maybe only one test assert the full output once to show structure

Makes sense, will try

> leaning towards no longer using beforeEach and instead just making helper functions

Yeah I see where you're coming from there. I think it would probably improve it in this case. I have seen the `beforeEach` pattern work well where it's use for arrange and act, and then the `it`s that follow it are just a line or two each of asserts.

Reply:
Okay, I've reworked this now based on the above. So we now parse each example feature with gherkin, and then assert on its output with references - so whilst we are still somewhat coupled to the hierarchical structure of a gherkin document, it will be an obvious compilation error if that changes, rather than assertions being out. Let me know what you think.

Reply:
Looks good. Thanks for the update